src/functions-and-arrays.js

```diff
@@ -1,73 +1,500 @@
-// Iteration #1: Find the maximum
+function maxOfTwoNumbers(n1, n2) {
+  if (n1 > n2) {
+    return n1;
+  } else if (n2 > n1) {
+    return n2;
+  } else {
+    return n1;
+  }
+}
 
 // Iteration #2: Find longest word
-const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
+const words = [
+  "mystery",
+  "brother",
+  "aviator",
+  "crocodile",
+  "pearl",
+  "orchard",
+  "crackpot"
+];
+
+function findLongestWord(words) {
+  let theLongesWord = "";
+  for (i = 0; i < words.length; i++) {
+    if (words[i].length > theLongesWord.length) {
+      theLongesWord = words[i];
+    }
+  }
+  if (theLongesWord == "") {
+    return null;
+  }
 
+  return theLongesWord;
+}
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumNumbers(numbers) {
+  let sumatory = 0;
+  for (i = 0; i < numbers.length; i++) {
+    sumatory += numbers[i];
+  }
+
+  return sumatory;
+}
+
+const mixedArr = [6, 12, "miami", 1, true, "barca", "200", "lisboa", 8, 10];
+
+function sum(mixedArr) {
+  let sumatory2 = 0;
+  for (i = 0; i < mixedArr.length; i++) {
+    if (typeof mixedArr[i] === "object") {
+      return "Unsupported data type sir or ma'am";
+    }
+    if (typeof mixedArr[i] === "boolean") {
+      if (mixedArr[i]) {
+        sumatory2 += 1;
+      }
+    } else if (typeof mixedArr[i] === "string") {
+      sumatory2 += mixedArr[i].length;
+    } else {
+      sumatory2 += mixedArr[i];
+    }
+  }
+  return sumatory2;
+}
+
 // Iteration #4: Calculate the average
+
 // Level 1: Array of numbers
-const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
+
+function averageNumbers(numbersAvg) {
+  console.log(numbersAvg);
+  let sumatory3 = 0;
+  for (i = 0; i < numbersAvg.length; i++) {
+    sumatory3 += numbersAvg[i];
+  }
+  if (sumatory3 == 0) {
+    return null;
+  }
+  sumy = sumatory3 / numbersAvg.length;
+  return sumy;
+}
 
 // Level 2: Array of strings
-const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
+function averageWordLength(wordsArr) {
+  let sumatory4 = 0;
+  for (i = 0; i < wordsArr.length; i++) {
+    sumatory4 += wordsArr[i].length;
+  }
+  if (sumatory4 == 0) {
+    return null;
+  }
+  sumy1 = sumatory4 / wordsArr.length;
```

Comment:
avoid creating a global variable that is being used by more than one function, and create that variable inside of the function instead